meerk40t/gui/scene/guicolors.py

```diff
@@ -0,0 +1,174 @@
+import wx
+
+def color_to_str(value):
+    temp = hex(value)
+    # The representation is backwards ABGR --> change
+    result = temp[:2] + temp[8:] + temp[6:8] + temp[4:6]+ temp[2:4]
+    # print("c2s from %s = %s" % (hex(value), result))
+    return result
+
+def str_to_color(value):
+    # The representation needs to be ABGR --> change
+    if len(value)<=8:
+        value = value + "FF" # append opacity
+    result = value[:2] + value[8:] + value[6:8] + value[4:6]+ value[2:4]
+    return int(result, base=16)
+
+class GuiColors():
+    '''
+    Provides and stores all relevant colors for a scene
+    '''
+    def __init__(self, context):
+        self.context = context
+        self.default_color = (
+            wx.Colour(0xA0, 0x7F, 0xA0),
+            wx.Colour(0x00, 0x00, 0xFF, 0x40),
+            wx.Colour(0xA0, 0xA0, 0xA0, 128),
+            wx.Colour(0x00, 0x00, 0x00),
+            wx.Colour("Grey"),
+            wx.Colour(0xFF, 0xA0, 0xA0, 0x60),
+            wx.Colour(0xA0, 0xA0, 0xA0, 0x40),
+            wx.Colour(0x00, 0xFF, 0x00, 0xA0),
+            wx.Colour(0xFF, 0x00, 0x00),
+            wx.Colour(0x00, 0xFF, 0x00),
+            wx.Colour(0x00, 0x00, 0xFF),
+            wx.Colour(0xFF, 0xFF, 0xFF)
+        )
+
+        self.context.setting(str, "color_manipulation", color_to_str(self.default_color[0].GetRGBA()))
+        self.context.setting(str, "color_laserpath", color_to_str(self.default_color[1].GetRGBA()))
+        self.context.setting(str, "color_grid", color_to_str(self.default_color[2].GetRGBA()))
+        self.context.setting(str, "color_guide", color_to_str(self.default_color[3].GetRGBA()))
+        self.context.setting(str, "color_background", color_to_str(self.default_color[4].GetRGBA()))
+        self.context.setting(str, "color_magnetline", color_to_str(self.default_color[5].GetRGBA()))
+        self.context.setting(str, "color_snap_visible", color_to_str(self.default_color[6].GetRGBA()))
+        self.context.setting(str, "color_snap_closeup", color_to_str(self.default_color[7].GetRGBA()))
+        self.context.setting(str, "color_selection1", color_to_str(self.default_color[8].GetRGBA()))
+        self.context.setting(str, "color_selection2", color_to_str(self.default_color[9].GetRGBA()))
+        self.context.setting(str, "color_selection3", color_to_str(self.default_color[10].GetRGBA()))
+        self.context.setting(str, "color_bed", color_to_str(self.default_color[11].GetRGBA()))
+
+    def set_default_colors(self):
+        '''
+        Reset all colors to default values...
+        '''
+        self.context("set color_manipulation %s" % color_to_str(self.default_color[0].GetRGBA()))
+        self.context("set color_laserpath %s" % color_to_str(self.default_color[1].GetRGBA()))
```

Comment:
Calling a bunch of calls to the console is a bit unusual for set. I guess there's not really anything wrong with it but it seems like you'd maybe just want to set the values these directly on `context.root`. They save the values during shutdown so whatever values are attached to the contexts should be persistent. The `setting()` is for loading either the persistent value or setting a default one. In theory everything should save if you did `self.context.color_laserpath = <str_value>`

Reply:
Done